File: java-backend/src/main/java/com/study/config/JwtProperties.java
package com.study.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * JWT 配置,绑定 jwt.* 。
 *
 * @ConfigurationProperties 把 application.yml / 环境变量里的 jwt 前缀配置绑定到这个对象。
 */
@ConfigurationProperties(prefix = "jwt")
public class JwtProperties {

    private String secret;
    private long expiresIn;

    public String getSecret() {
        return secret;
    }

    public void setSecret(String secret) {
        this.secret = secret;
    }

    public long getExpiresIn() {
        return expiresIn;
    }

    public void setExpiresIn(long expiresIn) {
        this.expiresIn = expiresIn;
    }
}

File: kotlin-backend/src/main/kotlin/com/study/config/JwtProperties.kt
package com.study.config

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * JWT 配置(构造器绑定):jwt.secret / jwt.expires-in 。
 *
 * @ConfigurationProperties 把 application.yml / 环境变量里的 jwt 前缀配置绑定到这个 data class。
 */
@ConfigurationProperties(prefix = "jwt")
data class JwtProperties(
    val secret: String,
    val expiresIn: Long,
)
